segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/SegmentState.java

```diff
@@ -74,17 +76,22 @@ public void serialize(DataOutputStream target) throws IOException {
      *
      * @param source The DataInputStream to deserialize from.
      * @return The deserialized SegmentState.
-     * @throws IOException If an exception occured.
+     * @throws IOException If an exception occurred.
+     * @throws DataCorruptionException If the given InputStream cannot be parsed into a SegmentState object.
      */
-    public static SegmentState deserialize(DataInputStream source) throws IOException {
-        byte version = source.readByte();
-        if (version == SERIALIZATION_VERSION) {
-            long segmentId = source.readLong();
-            String segmentName = source.readUTF();
-            Map<UUID, Long> attributes = AttributeSerializer.deserialize(source);
-            return new SegmentState(segmentId, segmentName, attributes);
-        } else {
-            throw new IOException(String.format("Unsupported version: %d.", version));
+    public static SegmentState deserialize(DataInputStream source) throws IOException, DataCorruptionException {
```

Comment:
Did you leave `throws IOException` on purpose? With the new code, we only throw `EOFException`, which extends ÌOException`, so the declaration is fine, but it might be better to reduce the scope.

Is there anything else that throws `IOException` here?

Reply:
Every operation from `DataInputStream` throws `IOException`. I only catch a subset of those, so I still need to leave `IOException` in the `throws` declaration.